test/spmd/tensor/test_api.py

```diff
@@ -0,0 +1,175 @@
+# Copyright (c) Meta Platforms, Inc. and affiliates
+import torch
+import torch.nn as nn
+from torch.testing._internal.common_utils import run_tests
+from spmd.test.common_utils import DistTensorTestBase, with_comms  # type: ignore
+from spmd import (
+    distribute_tensor,
+    distribute_module,
+    DeviceMesh,
+    DTensor,
+    Shard,
+    Replicate,
+)
+
+
+class MyModel(nn.Module):
+    def __init__(self, n_features, n_layers, device):
+        super().__init__()
+        self.seq = nn.Sequential(
+            *[
+                nn.Linear(n_features, n_features, device=device)
+                for _ in range(n_layers)
+            ]
+        )
+
+    def forward(self, x):
+        return self.seq(x)
+
+    def reset_parameters(self):
+        for m in self.seq:
+            m.reset_parameters()
+
+
+class DTensorAPITest(DistTensorTestBase):
+    @with_comms
+    def test_distribute_tensor(self):
+        device_mesh = DeviceMesh(self.device_type, list(range(self.world_size)))
+        shard_spec = [Shard(0)]
+        tensor_to_shard = torch.randn(12, 3)
+        sharded_tensor = distribute_tensor(
+            tensor_to_shard, device_mesh, shard_spec
+        )
+        self.assertEqual(sharded_tensor.size(), torch.Size([12, 3]))
+        local_tensor = sharded_tensor.to_local()
+        self.assertEqual(local_tensor.size(), torch.Size([3, 3]))
```

Comment:
Should we condition on the size of the device mesh? Looks like this assumes 4 GPUs?